clients/drcachesim/common/options.cpp

```diff
@@ -714,6 +714,18 @@ droption_t<std::string>
                    "Cache hierarchy configuration file",
                    "The full path to the cache hierarchy configuration file.");
 
+droption_t<bool>
+    op_enable_noise_generator(DROPTION_SCOPE_FRONTEND, "enable_noise_generator", false,
```

Comment:
But I might want multiple noise streams.  These end up looking like software threads and the scheduler will interleave them with real traced threads, right?  So to emulate a multi-tenant scenario we might want to add 100 different noise streams, 10 each for 10 different emulated processes and so each with different characteristics?  Just trying to plan ahead.

Reply:
Right, I guess just wanted to get to the simplest, working implementation for now.
But adding multiple ones is straightforward, so I did that.
The user can choose the number of "noise generator threads" with `-add_noise_generator N` (instead of using `-enable_noise_generator`).
However this begs the questions: how many threads per process? (for now each noise generator is a single thread in its own process) and how many records to generate for each? (for now they generate `-noise_generator_num_records` each).

Reply:
Obsolete.